src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,13 @@
+const initialState = [];
+const booksReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case 'ADD_BOOK':
+      return [...state, action.payload];
+    case 'REMOVE_BOOK':
+      return state.filter((book) => book.id !== action.payload);
+    default:
+      return state;
+  }
+};
```

Comment:
- You have done great so far :+1: :+1: But there are a few requirements you need to meet before moving to the next milestone. :+1: 
>Includes a reducer that adds a book
Includes a reducer that removes a book 

- [ ] As stated in the project requirement, you need to add the actions creators for adding and removing books. :+1: Currently you already have their cases in the reducers, you also need to create the action creator functions,. :+1: :smile:  :eg: 
```sh
export const addNewBook = (payload) => ({ type: ADD_BOOK, payload });
```

- [ ] Also at the very top of this file, you need to assign your action type to a variable and it needs to follow this pattern `const ADD_BOOK = 'books/booksSlice/ADD_BOOK';`. I'll suggest you update this too. :smile: :wave: :small_airplane: